spyder/utils/environ.py

```diff
@@ -22,11 +23,39 @@
 from qtpy.QtWidgets import QMessageBox
 
 # Local imports
-from spyder.config.base import _, running_in_ci
+from spyder.config.base import _, running_in_ci, get_conf_path
 from spyder.widgets.collectionseditor import CollectionsEditor
 from spyder.utils.icon_manager import ima
 from spyder.utils.programs import run_shell_command
 
+logger = logging.getLogger(__name__)
+
+if os.name == 'posix':
+    """
+    To get user environment variables from login and interactive startup files
+    on posix, both -l and -i flags must be used. Parsing the environment list
+    is problematic if the variable has a newline character, but Python's
+    os.environ can do this for us. However, executing Python in an interactive
+    shell in a subprocess causes Spyder to hang on Linux. Using a shell script
+    resolves the issue. Note that -i must be in the sha-bang line; if -i and
+    -l are swapped, Spyder will hang.
+    """
+    script_text = None
+    shell = os.getenv('SHELL', '/bin/bash')
+    user_env_script = Path(get_conf_path()) / 'user-env.sh'
```

Comment:
We could also write this script in the user temp directory. Why do you think it's better to write it in `conf_path`?

Reply:
I don't feel strongly one way or the other. My reasoning was that we don't have control over when a temp file is removed. Presumably, this will only occur when the user logs out or the system is rebooted. Since Spyder will only create the file once per session, but requires the file throughout the session, is there a risk that the file will be purged during a Spyder session?

Reply:
> I don't feel strongly one way or the other. My reasoning was that we don't have control over when a temp file is removed

Ok, I think it's fine to leave it in `conf_path` then.